hw10_program_optimization/stats.go

```diff
@@ -19,49 +18,50 @@ type User struct {
 	Address  string
 }
 
-type DomainStat map[string]int
+type DomainStat map[string]int32
 
 func GetDomainStat(r io.Reader, domain string) (DomainStat, error) {
-	u, err := getUsers(r)
+	e, err := getUsers(r, domain)
 	if err != nil {
-		return nil, fmt.Errorf("get users error: %s", err)
+		return nil, fmt.Errorf("get users error: %w", err)
 	}
-	return countDomains(u, domain)
+	return countDomains(e, domain), nil
 }
 
-type users [100_000]User
+type users []User
 
-func getUsers(r io.Reader) (result users, err error) {
-	content, err := ioutil.ReadAll(r)
-	if err != nil {
-		return
+func getUsers(r io.Reader, domain string) (result users, err error) {
+	scanner := bufio.NewScanner(r)
+	lines := make([]string, 0)
```

Comment:
https://github.com/uber-go/guide/blob/master/style.md#nil-is-a-valid-slice

Reply:
done